controllers/clustertopology_controller_compute_test.go

```diff
@@ -17,15 +17,325 @@ limitations under the License.
 )
 
+func TestGetClass(t *testing.T) {
+	fakeScheme := runtime.NewScheme()
+	_ = apiextensionsv1.AddToScheme(fakeScheme)
+	_ = clusterv1.AddToScheme(fakeScheme)
+
+	crds := []client.Object{
+		&apiextensionsv1.CustomResourceDefinition{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: apiextensionsv1.SchemeGroupVersion.String(),
+				Kind:       "CustomResourceDefinition",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "fakeinfrastructureclustertemplates.infrastructure.cluster.x-k8s.io",
+				Labels: map[string]string{
+					"cluster.x-k8s.io/v1alpha4": "v1alpha4_v1alpha5",
+				},
+			},
+			Spec: apiextensionsv1.CustomResourceDefinitionSpec{
+				Group: fakeInfrastructureProviderGroupVersion.Group,
+				Names: apiextensionsv1.CustomResourceDefinitionNames{
+					Kind: "FakeInfrastructureClusterTemplate",
+				},
+			},
+		},
+		&apiextensionsv1.CustomResourceDefinition{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: apiextensionsv1.SchemeGroupVersion.String(),
+				Kind:       "CustomResourceDefinition",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "fakecontrolplanetemplates.infrastructure.cluster.x-k8s.io",
+				Labels: map[string]string{
+					"cluster.x-k8s.io/v1alpha4": "v1alpha4",
+				},
+			},
+			Spec: apiextensionsv1.CustomResourceDefinitionSpec{
+				Group: fakeControlPlaneProviderGroupVersion.Group,
+				Names: apiextensionsv1.CustomResourceDefinitionNames{
+					Kind: "FakeControlPlaneTemplate",
+				},
+			},
+		},
+		&apiextensionsv1.CustomResourceDefinition{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: apiextensionsv1.SchemeGroupVersion.String(),
+				Kind:       "CustomResourceDefinition",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "fakeinfrastructuremachinetemplates.infrastructure.cluster.x-k8s.io",
+				Labels: map[string]string{
+					"cluster.x-k8s.io/v1alpha4": "v1alpha4_v1alpha5",
+				},
+			},
+			Spec: apiextensionsv1.CustomResourceDefinitionSpec{
+				Group: fakeInfrastructureProviderGroupVersion.Group,
+				Names: apiextensionsv1.CustomResourceDefinitionNames{
+					Kind: "FakeInfrastructureMachineTemplate",
+				},
+			},
+		},
+		&apiextensionsv1.CustomResourceDefinition{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: apiextensionsv1.SchemeGroupVersion.String(),
+				Kind:       "CustomResourceDefinition",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "fakebootstraptemplates.infrastructure.cluster.x-k8s.io",
+				Labels: map[string]string{
+					"cluster.x-k8s.io/v1alpha4": "v1alpha4",
+				},
+			},
+			Spec: apiextensionsv1.CustomResourceDefinitionSpec{
+				Group: fakeBootstrapProviderGroupVersion.Group,
+				Names: apiextensionsv1.CustomResourceDefinitionNames{
+					Kind: "FakeBoostrapTemplate",
+				},
+			},
+		},
+	}
+
+	infraClusterTemplate := newFakeInfrastructureClusterTemplate(metav1.NamespaceDefault, "infraclustertemplate1").Obj()
+	infraClusterTemplatev1alpha5 := newFakeInfrastructureClusterTemplate(metav1.NamespaceDefault, "infraclustertemplate1").Obj()
```

Comment:
s/v1alpha5/nNext
So we don't have to update when the next release will be generated

Reply:
I'll change it to v99, because it has to be comparable via `CompareKubeAwareVersionStrings`. vNext is parsed as: major: 0 and minor: 0 and is thous < v1alpha4